lib/inspec/backend.rb

```diff
@@ -53,19 +53,20 @@ def self.create(config)
         raise "Can't connect to transport backend '#{name}'."
       end
 
+      # set caching settings
+      if config[:backend_cache]
+        connection.enable_cache(:command)
+      elsif config[:debug_shell]
+        connection.disable_cache(:file)
+        connection.disable_cache(:command)
+      elsif !config[:backend_cache] && config[:backend] != :mock
```

Comment:
Could you explain why this can't be an `else`? It's not quite clear to me. And probably warrants a code comment explaining the condition (i.e. why the mock backend is special here).

Reply:
Yeah I'll add a comment. Basically the mock transport uses the base connection caching also. So we never want to disable it. We hard set the cache values for testing and usually bypass thor so it would always be false. It actually may be cleaner to put the mock check on the enable and just ensure its enabled.

Reply:
What would happen in the mock transport if it was disabled? Slower tests? If so, by how much?

Reply:
Not really, you cant actually run file/command calls via mock without preloading it. Else it will just fall to errors. So for the core of MockTransport to work caching is always needed. Ex:
https://github.com/chef/train/blob/master/lib/train/transports/mock.rb#L139